src/nginx.py

```diff
@@ -344,16 +255,63 @@ def _basic_auth(self, enabled: bool) -> List[Optional[Dict[str, Any]]]:
             ]
         return []
 
-    def _server(self, addresses_by_role: Dict[str, Set[str]], tls: bool = False) -> Dict[str, Any]:
+    def _listen(self, port: int, ssl: bool, http2: bool) -> List[Dict[str, Any]]:
+        directives = []
+        directives.append(
+            {"directive": "listen", "args": self._listen_args(port, False, ssl, http2)}
+        )
+        directives.append(
+            {"directive": "listen", "args": self._listen_args(port, True, ssl, http2)}
+        )
+        return directives
+
+    def _listen_args(self, port: int, ipv6: bool, ssl: bool, http2: bool) -> List[str]:
+        args = []
+        if ipv6:
+            args.append(f"[::]:{port}")
+        else:
+            args.append(f"{port}")
+        if ssl:
+            args.append("ssl")
+        if http2:
+            args.append("http2")
+        return args
+
+    def _servers(
+        self, addresses_by_role: Dict[str, Set[str]], tls: bool = False
+    ) -> List[Dict[str, Any]]:
+        servers = []
+        roles = addresses_by_role.keys()
+
+        if "distributor" in roles or "all" in roles:
```

Comment:
can we extract those literals as constants somewhere

Reply:
Simplified the code significantly by using values from enums/dicts from `Tempo` where it was possible. Now it should look better :)